model_handler/views.py

```
@@ -0,0 +1,448 @@
+            messages.error(self.request, f"Error processing request: {str(e)}")
+            return self.form_invalid(form)
+    
+    def form_invalid(self, form):
+        messages.error(self.request, "Invalid form data. Please check your inputs.")
+        return super().form_invalid(form)
+    
+    def post(self, request, *args, **kwargs):
+        # Handle AJAX requests
+        if request.headers.get('x-requested-with') == 'XMLHttpRequest':
+            try:
+                data = json.loads(request.body)
```

Ruff (0.8.2)

Undefined name `json`

(F821)